dias/analyzers/source_spectra_analyzer.py

```diff
@@ -0,0 +1,733 @@
+            for file_index, files in enumerate(all_files):
+                # Read file time range
+                with h5py.File(files, "r") as handler:
+                    file_timestamp = handler["index_map"]["time"]["ctime"][:]
+
+                csd = int(np.median(ephemeris.csd(file_timestamp)))
+
+                # Compute source coordinates
+                src_ra, src_dec = ephemeris.object_coords(
+                    src_body, date=np.median(file_timestamp), deg=False
+                )
+                source_start_time, source_stop_time = file_list[file_index][1]
+
+                is_daytime = 0.0
+                # test if the source is transiting in the daytime
+                solar_rise = ephemeris.solar_rising(
+                    source_start_time - 24.0 * 3600.0, end_time=source_stop_time
+                )
+                for rr in solar_rise:
+                    ss = ephemeris.solar_setting(rr)[0]
+                    if (source_start_time <= ss) and (rr <= source_stop_time):
+                        is_daytime += 1
+                        tt = ephemeris.solar_transit(rr)[0]
+                        # if Sun is in the beam
+                        if source_start_time <= tt <= source_stop_time:
+                            is_daytime += 1
+                        break
+
+                if is_daytime > 1:
+                    self.logger.info(
+                        "Not processing %s as Sun is in the primary beam" % (src)
+                    )
+                    continue
+
+                self.logger.info("Now processing %s transit on CSD %d" % (src, csd))
+
+                # Look up inputmap
+                inputmap = tools.get_correlator_inputs(
+                    ephemeris.unix_to_datetime(ephemeris.csd_to_unix(csd)),
+                    correlator=self.correlator,
+                )
+
+                # Load index map and reverse map
+                data = andata.CorrData.from_acq_h5(
+                    files,
+                    start=int(np.argmin(np.abs(file_timestamp - source_start_time))),
+                    stop=int(np.argmin(np.abs(file_timestamp - source_stop_time))),
+                    datasets=["reverse_map", "flags/inputs"],
+                    apply_gain=False,
+                    renormalize=False,
+                )
+                # Determine axes
+                nfreq = data.nfreq
+                nblock = int(np.ceil(nfreq / float(self.nfreq_per_block)))
+
+                data_timestamp = data.time  # 1D array
+                ntime = data_timestamp.size
+
+                # Get baselines
+                prod, _, dist, _, _, scale = self.get_baselines(
+                    data.index_map, inputmap
+                )
+
+                # Determine groups
+                pols = np.array(sorted(prod.keys()))
+                npol = pols.size
+
+                # Calculate counts
+                cnt = np.zeros((data.index_map["stack"].size, ntime), dtype=np.float32)
```

Comment:
I'm not sure if draco is a requirement for Dias.  If it is, there is a cython function for calculating this count variable in `draco.util.tools`.  It would be called as:

```
cnt = tools.calculate_redundancy(data.input_flags[:], data.prod[:], data.reverse_map["stack"]["stack"][:], data.nstack)
```

Reply:
It is not. But it should be fine to add it.

Reply:
I went ahead and updated the analyzer to use the function in draco, and also added draco as a dependency to Dias.  It is much faster, for a typical window around source transit it takes 2 seconds compared to 230 seconds.

Reply:
It looks like that has caused a build error in Travis CI due to mpi4py, which is a draco dependency.

Reply:
As suggested by @nritsche, adding `libopenmpi-dev` and `openmpi-bin` under `apt: packages:` in `.travis.yml` fixed this.